midx.c

```diff
@@ -1081,15 +1096,16 @@ int verify_midx_file(struct repository *r, const char *object_dir, unsigned flag
 {
 	struct pair_pos_vs_id *pairs = NULL;
 	uint32_t i;
-	struct progress *progress;
+	struct progress *progress = NULL;
```

Comment:
This commit is a bit big. It adds progress to "write", "verify", and "expire" subcommands. These could be split into separate commits, so review is easier. There will be less to complain about per patch, so the discussion will be easier to understand.

To split the commit, my usual workflow is to:

1.  `git rebase -i <base>` and change the commit to `edit` instead of `pick`.
2. `git reset HEAD~1`. This will "erase" your current commit, but leave the changes.
3. `git add -p` to add only the parts needed for one commit.
4. `git commit -s` to commit those changes, adding a detailed message.
5. Repeat 3 and 4 until done with the current changes.
6. `git rebase --continue` to get the rest of the series.